pytng/pytng.pyx

```diff
@@ -267,33 +309,21 @@ cdef class TNGFile:
         else:
             time = frame_time * 1e12
 
+        # BOX SHAPE
+        cdef MemoryWrapper wrap_box
+        cdef float* box_s
+        wrap_box = MemoryWrapper(3 * 3 * sizeof(float))
+        box_shape = <float*> wrap_box.ptr
         cdef np.ndarray[ndim=2, dtype=np.float32_t, mode='c'] box = np.empty((3, 3), dtype=np.float32)
-        cdef char data_type
-        cdef void* box_shape = NULL
-        cdef float* float_box
-        cdef double* double_box
-
-        try:
-            ok = tng_data_vector_interval_get(self._traj, TNG_TRAJ_BOX_SHAPE, self.step, self.step, TNG_USE_HASH,
-                                              &box_shape, &stride_length, &n_values_per_frame, &data_type)
-            if ok != TNG_SUCCESS:
-                raise IOError("error reading box shape")
-
-            if data_type == TNG_DOUBLE_DATA:
-                double_box = <double*>box_shape
-                for j in range(3):
-                    for k in range(3):
-                        box[j, k] = double_box[j*3 + k]
-            else:
-                float_box = <float*>box_shape
-                for j in range(3):
-                    for k in range(3):
-                        box[j, k] = float_box[j*3 + k]
-            box *= self.distance_scale
-        finally:
-            if box_shape != NULL:
-                free(box_shape)
+        ok = tng_util_box_shape_read_range(self._traj, self.step, self.step, &box_shape, &stride_length) #TODO this will break when using frames spaced more than 1 apart
```

Comment:
so reading the tng docs for `data_vector_interval_get` it says it'll do the malloc for you, and you're responsible for freeing it.  Is it instead that it'll `realloc` if it's not the right size?

Reply:
I'll have a look.